include/text.php

```diff
@@ -1169,23 +1169,32 @@ function redir_private_images($a, &$item)
 	}
 }
 
+function body_with_cw($item)
+{
+	// Add the content warning
+	if (!empty($item['content-warning'])) {
+		$body = $item['content-warning'] . '[spoiler]' . $item["body"] . '[/spoiler]';
+	} else {
+		$body = $item["body"];
+	}
+	return $body;
+}
+
 function put_item_in_cache(&$item, $update = false)
 {
 	$rendered_hash = defaults($item, 'rendered-hash', '');
 
 	if ($rendered_hash == ''
 		|| $item["rendered-html"] == ""
-		|| $rendered_hash != hash("md5", $item["body"])
+		|| $rendered_hash != hash("md5", body_with_cw($item))
 		|| Config::get("system", "ignore_cache")
 	) {
 		// The function "redir_private_images" changes the body.
 		// I'm not sure if we should store it permanently, so we save the old value.
 		$body = $item["body"];
 
 		// Add the content warning
-		if (!empty($item['content-warning'])) {
```

Comment:
You can just move this code block outside the rendered hash condition without having to create another method.

Reply:
No. When I move it to another place, this will have side effects, since the array is called by reference.

Reply:
I believe this will clear any confusion: #4598